datajoint/erd.py

```diff
@@ -206,8 +206,8 @@ def _make_graph(self):
                 nx.algorithms.boundary.node_boundary(nx.DiGraph(self).reverse(), self.nodes_to_show))
             nodes = self.nodes_to_show.union(a for a in gaps if a.isdigit)
             # construct subgraph and rename nodes to class names
-            graph = nx.DiGraph(self).subgraph(nodes)
-            nx.set_node_attributes(graph, 'node_type', {n: _get_tier(n) for n in graph})
+            graph = nx.DiGraph(nx.DiGraph(self).subgraph(nodes))
```

Comment:
what's the reason for the redundancy?

Reply:
The call now returns a subgraph 'view', which is immutable and cannot be relabeled later on in line 219's ```nx.relabel_nodes(graph, mapping, copy=False)```. I also tried doing a ```return nx.relabel_nodes(graph, mapping, copy=True)``` in that line but it didn't work on 1st pass - from here I didn't really investigate the API change further - since copy=True imples making a copy and the line you are mentioning would also makes a a copy I viewed these as equivalent and moved on..

Reply:
Yes I'm imagining that `self.subgraph` should suffice. @dimitri-yatsenko do you recall why we have to do `nx.Digraph(self).subgraph`? I have a feeling that this is legacy code

Reply:
working on the speedtest now - it seems the constructor is called internally and the ```__init___``` signatures between the classes vary, though I haven't dug further just yet & theoretically calling via superclass would work around this..

Reply:
it seems that networkx's ```digraph.py`` calls H.__class__() to create an instance copy within the ```subgraph``` function - since this refers to ```Dependencies``` which requires ```connection``` as a parameter, the call fails. Making this an optional argument allows ```self.subgraph(...)``` to work, however, this could potentially be undesirable w/r/t the rest of the code depending on opinions...

Reply:
We probably should override `subgraph` so `connection` gets passed down - although it also makes sense to check if `connection` is needed in the subgraphs.

Reply:
Thinking to audit for connection use and proceed or just take the performance hit -

Maintaining a random graph related function in-tree to avoid this issue doesn't seem architecturally correct since this class isn't *really* a 'graph subclass' in terms of 'providing additional graph functionality' (even though it is *technically* a subclass) .. 

That said it isn't such a big deal to copy the function in, annotate the reasoning and sync when needed with upstream. Networkx is BSD licencsed btw so this wouldn't impact DJ.

@dimitri-yatsenko care to weigh in?

Reply:
If everything works, then no need to pass the connection

Reply:
@dimitri-yatsenko apologies if unclear - so does this mean the additional patch to ```datajoint/dependencies.py ``` mentioned in the comments here can be added to this PR after comment cleanup?